<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GeneticData • compbio2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GeneticData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">3.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00_Syllabus_and_Expectations.html">Syllabus</a>
    </li>
    <li>
      <a href="../articles/01_Getting_Started_with_R.html">Starting with R</a>
    </li>
    <li>
      <a href="../articles/02_Starting_with_Data.html">Starting with data</a>
    </li>
    <li>
      <a href="../articles/03_Manipulating_Data.html">Manipulating, analyzing and exporting data with tidyverse</a>
    </li>
    <li>
      <a href="../articles/04-plotting.html">Data visualization with ggplot2</a>
    </li>
    <li>
      <a href="../articles/05_Functions.html">Functions and repeatability</a>
    </li>
    <li>
      <a href="../articles/06_Exploration_Setup.html">Exploration</a>
    </li>
    <li>
      <a href="../articles/06_Project_Structuring.html">Project Structuring</a>
    </li>
    <li>
      <a href="../articles/08_linear_models.html">Introduction to Linear Models: Building and Interpreting Linear Regressions</a>
    </li>
    <li>
      <a href="../articles/09_Tree_of_life.html">GeneticData</a>
    </li>
    <li>
      <a href="../articles/10_GBIF_and_Location.html">GBIF And Spatial Data</a>
    </li>
    <li>
      <a href="../articles/11_iNEXT.html">Camera Trap Data Analysis</a>
    </li>
    <li>
      <a href="../articles/bonus_tables.html">Tables.Rmd</a>
    </li>
    <li>
      <a href="../articles/homework_0.html">How to Do Homework</a>
    </li>
    <li>
      <a href="../articles/homework_1.html">Homework One</a>
    </li>
    <li>
      <a href="../articles/homework_2.html">Homework Two</a>
    </li>
    <li>
      <a href="../articles/homework_3.html">Homework Three</a>
    </li>
    <li>
      <a href="../articles/homework_4.html">Homework Four</a>
    </li>
    <li>
      <a href="../articles/homework_5.html">Homework Five: Functions</a>
    </li>
    <li>
      <a href="../articles/homework_6.html">HomeworkSix.Rmd</a>
    </li>
    <li>
      <a href="../articles/homework_7.html">Homework Seven</a>
    </li>
    <li>
      <a href="../articles/HomeworkFive.html">Homework Five</a>
    </li>
    <li>
      <a href="../articles/HomeworkSix.html">HomeworkSix</a>
    </li>
    <li>
      <a href="../articles/project_final.html">Project: Final</a>
    </li>
    <li>
      <a href="../articles/project_one.html">ProjectOne</a>
    </li>
    <li>
      <a href="../articles/project_three.html">ProjectThree</a>
    </li>
    <li>
      <a href="../articles/project_two.html">Project Two</a>
    </li>
    <li>
      <a href="../articles/Simulation.html">Simulations</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GeneticData</h1>
            
      

      <div class="hidden name"><code>09_Tree_of_life.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="a-quick-wrap-up-of-the-git-stuff-from-last-time">A quick wrap-up of the git stuff from last time<a class="anchor" aria-label="anchor" href="#a-quick-wrap-up-of-the-git-stuff-from-last-time"></a>
</h2>
<p>We’ll install a package called <code>usethis</code>, which contains
some helpful programming tools for the weeks ahead:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"usethis"</span><span class="op">)</span></span></code></pre></div>
<p>We’ll use this to create a personal access token for github. This
ensures we don’t have to retype our password every time we need to
access GitHub.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/github-token.html" class="external-link">create_github_token</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p>And we will use this token to access our account via RStudio.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">credentials</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/credentials/reference/set_github_pat.html" class="external-link">set_github_pat</a></span><span class="op">(</span><span class="st">"PAT here"</span><span class="op">)</span></span></code></pre></div>
<p>Now, refresh your R session.</p>
<p>Once this is done, we can check that this worked like so:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/git_sitrep.html" class="external-link">git_sitrep</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-tree-of-life">The Tree of Life<a class="anchor" aria-label="anchor" href="#the-tree-of-life"></a>
</h2>
<p>First, we need to do some installs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/rotl"</span><span class="op">)</span></span>
<span><span class="co">#if (!requireNamespace("BiocManager"))</span></span>
<span><span class="co">#    install.packages("BiocManager")</span></span>
<span><span class="co">#BiocManager::install()</span></span>
<span><span class="co"># Install package dependencies</span></span>
<span><span class="co">#BiocManager::install("Biostrings")</span></span>
<span><span class="co">#BiocManager::install("biomaRt")</span></span>
<span><span class="co">#install.packages("biomartr", dependencies = TRUE)</span></span></code></pre></div>
<p>Now, load your packages:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rotl/" class="external-link">rotl</a></span><span class="op">)</span></span>
<span><span class="co">#library(biomatr)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="open-tree">Open Tree<a class="anchor" aria-label="anchor" href="#open-tree"></a>
</h2>
<p>First, we’re going to look for some data. Discuss with your neighbor
a clade you like and want to build a tree of. For example, perhaps I
really like hominids.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rotl/" class="external-link">rotl</a></span><span class="op">)</span></span>
<span><span class="va">apes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pongo"</span>, <span class="st">"Pan"</span>, <span class="st">"Gorilla"</span>, <span class="st">"Hoolock"</span>, <span class="st">"Homo"</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">resolved_names</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tnrs_match_names.html" class="external-link">tnrs_match_names</a></span><span class="op">(</span><span class="va">apes</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   search_string unique_name approximate_match score ott_id is_synonym</span></span>
<span><span class="co">## 1         pongo       Pongo             FALSE     1 417949      FALSE</span></span>
<span><span class="co">## 2           pan         Pan             FALSE     1 417957      FALSE</span></span>
<span><span class="co">## 3       gorilla     Gorilla             FALSE     1 417969      FALSE</span></span>
<span><span class="co">## 4       hoolock     Hoolock             FALSE     1 712902      FALSE</span></span>
<span><span class="co">## 5          homo        Homo             FALSE     1 770309      FALSE</span></span>
<span><span class="co">##                               flags number_matches</span></span>
<span><span class="co">## 1                                                2</span></span>
<span><span class="co">## 2                    sibling_higher              2</span></span>
<span><span class="co">## 3                    sibling_higher              1</span></span>
<span><span class="co">## 4                                                1</span></span>
<span><span class="co">## 5 extinct_inherited, sibling_higher              1</span></span></code></pre>
<p>Put the names of five genera into the function. If you want, you can
use mine below, or pick new ones you prefer.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Martialis"</span>, <span class="st">"Atta"</span>, <span class="st">"Ectatomma"</span>, <span class="st">"Tatuidris"</span>, <span class="st">"Aneuretus"</span>, <span class="st">"Xymmer"</span>, <span class="st">"Aenictus"</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">resolved_names</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tnrs_match_names.html" class="external-link">tnrs_match_names</a></span><span class="op">(</span><span class="va">ants</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   search_string unique_name approximate_match score  ott_id is_synonym flags</span></span>
<span><span class="co">## 1     martialis   Martialis             FALSE     1  334634      FALSE      </span></span>
<span><span class="co">## 2          atta        Atta             FALSE     1  709964      FALSE      </span></span>
<span><span class="co">## 3     ectatomma   Ectatomma             FALSE     1  162845      FALSE      </span></span>
<span><span class="co">## 4     tatuidris   Tatuidris             FALSE     1  150177      FALSE      </span></span>
<span><span class="co">## 5     aneuretus   Aneuretus             FALSE     1  425343      FALSE      </span></span>
<span><span class="co">## 6        xymmer      Xymmer             FALSE     1 4583488      FALSE      </span></span>
<span><span class="co">## 7      aenictus    Aenictus             FALSE     1  169919      FALSE      </span></span>
<span><span class="co">##   number_matches</span></span>
<span><span class="co">## 1              1</span></span>
<span><span class="co">## 2              1</span></span>
<span><span class="co">## 3              1</span></span>
<span><span class="co">## 4              1</span></span>
<span><span class="co">## 5              1</span></span>
<span><span class="co">## 6              1</span></span>
<span><span class="co">## 7              1</span></span></code></pre>
<p>Try to view them on a tree:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tol_induced_subtree.html" class="external-link">tol_induced_subtree</a></span><span class="op">(</span>ott_ids <span class="op">=</span> <span class="fu"><a href="https://docs.ropensci.org/rotl/reference/taxonomy-methods.html" class="external-link">ott_id</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Progress [----------------------------------] 0/43 (  0%) ?sProgress [================================] 43/43 (100%)  0s                                                            </span></span></code></pre>
<pre><code><span><span class="co">## Warning in collapse_singles(tr, show_progress): Dropping singleton nodes with</span></span>
<span><span class="co">## labels: mrcaott1682ott35311, mrcaott1682ott34821, mrcaott34821ott950983,</span></span>
<span><span class="co">## Aneuretinae ott393261, mrcaott4706ott5525, mrcaott5525ott8038,</span></span>
<span><span class="co">## mrcaott8038ott8685, mrcaott8685ott36581, mrcaott8685ott56535,</span></span>
<span><span class="co">## mrcaott8685ott72729, mrcaott8685ott86888, mrcaott86888ott129302, Attini</span></span>
<span><span class="co">## ott344333, mrcaott114767ott114779, mrcaott114779ott339738,</span></span>
<span><span class="co">## mrcaott114779ott116619, mrcaott116619ott148996, mrcaott148996ott1071522,</span></span>
<span><span class="co">## mrcaott148996ott150184, mrcaott148996ott339763, mrcaott148996ott409299,</span></span>
<span><span class="co">## mrcaott48019ott688185, Ectatomminae ott327970, mrcaott48019ott162057,</span></span>
<span><span class="co">## mrcaott7438ott5553505, mrcaott7438ott7439, mrcaott7439ott165217,</span></span>
<span><span class="co">## mrcaott7439ott15430, mrcaott15430ott120483, mrcaott15430ott150187,</span></span>
<span><span class="co">## mrcaott15430ott169913, mrcaott17752ott98880, mrcaott17752ott150176,</span></span>
<span><span class="co">## mrcaott150176ott1037289, mrcaott150176ott4539262, Agroecomyrmecinae ott504610,</span></span>
<span><span class="co">## mrcaott98886ott152924, mrcaott152924ott452719, mrcaott152924ott739132,</span></span>
<span><span class="co">## mrcaott152924ott503313, mrcaott503313ott928530, mrcaott928530ott6280946,</span></span>
<span><span class="co">## Martialinae ott334635</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span></span></code></pre></div>
<p><img src="09_Tree_of_life_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>I’m a little surprised by this tree.</p>
</div>
<div class="section level2">
<h2 id="looking-for-data">Looking for data<a class="anchor" aria-label="anchor" href="#looking-for-data"></a>
</h2>
<p>We’re going to try to build a tree of these taxa from GenBank Data.
Much like accessing NOAA data, we will need to make an API Key. Go <a href="https://www.ncbi.nlm.nih.gov/account/" class="external-link">here</a> and register.</p>
<p>Now that we have that key, we can access NCBI:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/" class="external-link">rentrez</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="st">"Ectatomma"</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Entrez search result with 707 hits (object contains 20 IDs and no web_history object)</span></span>
<span><span class="co">##  Search term (as translated):  "Ectatomma"[Organism] OR Ectatomma[All Fields]</span></span></code></pre>
<p>Holy moley that’s a lot of data. If we do that for every taxon, we’ll
end up with a mess. Why don’t we try to filter it a bit.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ectatomma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="st">"Ectatomma AND 2018:2019[PDAT]"</span>, retmax <span class="op">=</span> <span class="fl">100</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></span></code></pre></div>
<p>So there is a good amount of data for our Ectatomma ants. We’re going
to download some. And this gets a little tricky. The first thing we will
do is try the <code>entrez_fetch</code> command:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"nuccore"</span>, id <span class="op">=</span> <span class="va">Ectatomma</span><span class="op">$</span><span class="va">ids</span>,</span>
<span>             rettype<span class="op">=</span><span class="st">"fasta"</span>, retmax<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "&gt;MK759606.1 Ectatomma ruidum voucher YB-BCI64481 cytochrome oxidase subunit 1 (COI) gene, partial cds; mitochondrial\nATATTTCATCTTTGCTATCTGAACTGGTTTAATTGGTTCATCTATAAGTATAATTATTCGATTAGGGCTA\nGGAACTTGCGGTTCTATTATTAATAATGACCAAGTTTTTAATTCTATTGTTACAGCTCATGCATTTATTA\nTAATTTTTTTCATGGTTATACCATTTATAATTGGAGGATTTGGAAATTTTTTAGTGCCTTTAATATTAGG\nATCTCCTGATATAGCCTACCCACGAATAAATAATATAAGATTTTGACTCCTTCCTCCCTCTATTATACTT\nTTAACTTTAAGAAGATTAATTAGTAGAGGAGTGGGAACAGGATGAACAGTTTATCCTCCCCTCGCATCTA\nGTGTTTTCCATAGAGGTTCGTCTGTTGATCTAGCTATTTTTTCATTACATATTGCAGGAATATCATCAAT\nTTTAGGGGCTACTAATTTTATTTCAACTATTATAAATATACACCATAAAAATATAACTATAGATAAAATT\nCCTCTTTTAGTTTGAGCTATTATAATTATTGCAATTTTACTACTTTTATCGTTACCTGTATTAGCTGGTG\nCTATTACAATACTTTTAACTGACCGAAATTTAAATACTTTATTCTTTGATCCTTCAGGAAGTGGGGATCC\nTATTCTTTATCAACATTTATTT\n\n"</span></span></code></pre>
<p>Look at the output - does it look like we have output for every
ID?</p>
<p>Let’s try the below. If you’ve forgotten, this is called a
<code>for loop</code>. It means that for each entry in the ids vector,
we will download a sequence from GenBank.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span><span class="op">(</span> <span class="va">id</span> <span class="kw">in</span> <span class="va">Ectatomma</span><span class="op">$</span><span class="va">ids</span><span class="op">)</span> <span class="op">{</span></span>
<span>     <span class="va">recs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"nuccore"</span>, id <span class="op">=</span> <span class="va">id</span>,</span>
<span>                          rettype<span class="op">=</span><span class="st">"fasta"</span>, retmax<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">recs</span>, file<span class="op">=</span><span class="st">"Ect.seqs.fasta"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span> <span class="op">}</span></span></code></pre></div>
<p>Now, let’s look at this file. What do we notice about it? What
information is in this file?</p>
<p>Before we move on, we’re going to try one other thing. Try modifying
one of the two chunks above (HINT: you will have to rerun both the
search and the fetch) in order to only get data from the mitochondrial
COI gene.</p>
<p>All right, let’s move on to trying to write a function that will
return all IDs for all of the ants.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ant</span>, <span class="st">" AND "</span>, <span class="st">"2016:2019[PDAT]"</span><span class="op">)</span></span>
<span>    <span class="va">search_returns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">20</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>Have a look at ?sapply. Talk to a partner and see if you can figure
out what sapply does.</p>
<p>So what is the problem with this output? It’s hard to parse - let’s
make it easier to pull data for one taxon at a time.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">search_returns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ant</span>,<span class="st">" AND "</span>, <span class="st">"2019:2022[PDAT]"</span><span class="op">)</span></span>
<span>    <span class="va">search_returns</span><span class="op">[</span><span class="va">ant</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">20</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p>There’s one remaining problem, though. Have a look at your outputs.
How do you think we can fix this one? Let’s have a look at when papers
related to Xymmer were published:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pmc"</span>, term<span class="op">=</span><span class="st">"Xymmer"</span>, id<span class="op">=</span> <span class="st">"1103332937"</span>, retmax<span class="op">=</span><span class="fl">20</span>, rettype<span class="op">=</span> <span class="st">"text"</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></span></code></pre></div>
<p>Run this as-is and take a look at the outputs.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">search_returns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span>    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ant</span>, <span class="st">"[Organism]"</span>, <span class="st">" AND 18s"</span><span class="op">)</span></span>
<span>    <span class="va">search_returns</span><span class="op">[</span><span class="va">ant</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">1</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span></span>
<span></span>
<span><span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p>Now, we’re going to get the genetic data.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">search_seq</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">labels</span><span class="op">)</span><span class="op">{</span></span>
<span>     <span class="va">recs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"nuccore"</span>, id <span class="op">=</span> <span class="va">labels</span>,</span>
<span>                          rettype<span class="op">=</span><span class="st">"fasta"</span>, retmax<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span>     <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">recs</span>, file<span class="op">=</span><span class="st">"Ants.seqs.fasta"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">recs</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="va">seq_returns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">labels</span>, <span class="va">search_seq</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="excercise">Excercise:<a class="anchor" aria-label="anchor" href="#excercise"></a>
</h2>
<p>18s is an easy one. Let’s try and pull another common gene for
phylogenetics, COX1.</p>
<p>Make a new cell and modify our search_year code to pull COX1 data.
What are the most immediate problems you notice?</p>
<p>Try resolving the issue. Can you pinpoint which part of our query
gives us trouble?</p>
</div>
<div class="section level2">
<h2 id="just-for-fun">Just for fun<a class="anchor" aria-label="anchor" href="#just-for-fun"></a>
</h2>
<p>Take a quick look at this output - it’s a little neat. We’re not
going to go far into building phylogenetic trees - that’s a topic for
Systematics. Here’s a quick look at these data.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"ape"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## The downloaded binary packages are in</span></span>
<span><span class="co">##  /var/folders/54/9kd8nf1x4fnft0ymvb11qmc80000gn/T//RtmptmmoSa/downloaded_packages</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s18</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.dna.html" class="external-link">read.dna</a></span><span class="op">(</span><span class="st">"Ants.seqs.fasta"</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></span></code></pre></div>
<p>These data are referred to as “unaligned” - this means we haven’t
assigned homology at the molecular level. The process of doing this is
called sequence alignment. Let’s take a look at doing that. We’ll
install a package called <code>msa</code>, for Multiple Sequence
Alignment.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"msa"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 'getOption("repos")' replaces Bioconductor standard repositories, see</span></span>
<span><span class="co">## 'help("repositories", package = "BiocManager")' for details.</span></span>
<span><span class="co">## Replacement repositories:</span></span>
<span><span class="co">##     CRAN: https://cran.rstudio.com/</span></span></code></pre>
<pre><code><span><span class="co">## Bioconductor version 3.19 (BiocManager 1.30.25), R 4.4.1 (2024-06-14)</span></span></code></pre>
<pre><code><span><span class="co">## Warning: package(s) not installed when version(s) same as or greater than current; use</span></span>
<span><span class="co">##   `force = TRUE` to re-install: 'msa'</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/UBod/msa" class="external-link">msa</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: Biostrings</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, setdiff, table,</span></span>
<span><span class="co">##     tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: XVector</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biostrings'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     strsplit</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span></code></pre></div>
<p>The process of assessing homology at the molecular level typically
involves a model of sequence evolution.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span></span>
<span><span class="va">mySequence</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readDNAStringSet</a></span><span class="op">(</span><span class="st">"Ants.seqs.fasta"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in .Call2("fasta_index", filexp_list, nrec, skip, seek.first.rec, :</span></span>
<span><span class="co">## reading FASTA file Ants.seqs.fasta: ignored 4586 invalid one-letter sequence</span></span>
<span><span class="co">## codes</span></span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">myAlignment</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msa/man/msa.html" class="external-link">msa</a></span><span class="op">(</span><span class="va">mySequence</span>, <span class="st">"ClustalOmega"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## using Gonnet</span></span></code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output_seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/msa/man/msaConvert.html" class="external-link">msaConvert</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">myAlignment</span>, type <span class="op">=</span> <span class="st">"phangorn::phyDat"</span><span class="op">)</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
