<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>GeneticData • compbio2020</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="GeneticData">
<meta property="og:description" content="compbio2020">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">compbio2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/00_Syllabus_and_Expectations.html">Syllabus</a>
    </li>
    <li>
      <a href="../articles/01_Getting_Started_with_R.html">Starting with R</a>
    </li>
    <li>
      <a href="../articles/02_Starting_with_Data.html">Starting with data</a>
    </li>
    <li>
      <a href="../articles/03_Manipulating_Data.html">Manipulating, analyzing and exporting data with tidyverse</a>
    </li>
    <li>
      <a href="../articles/04-plotting.html">Data visualization with ggplot2</a>
    </li>
    <li>
      <a href="../articles/05_Functions.html">Functions and repeatability</a>
    </li>
    <li>
      <a href="../articles/06_Exploration_Setup.html">Exploration</a>
    </li>
    <li>
      <a href="../articles/08_linear_models.html">Introduction to Linear Models: Building and Interpreting Linear Regressions</a>
    </li>
    <li>
      <a href="../articles/09_Tree_of_life.html">GeneticData</a>
    </li>
    <li>
      <a href="../articles/10_GBIF_and_Location.html">GBIF And Spatial Data</a>
    </li>
    <li>
      <a href="../articles/HomeworkFive.html">Homework Five</a>
    </li>
    <li>
      <a href="../articles/HomeworkFour.html">HomeworkFour</a>
    </li>
    <li>
      <a href="../articles/HomeworkSix.html">HomeworkSix</a>
    </li>
    <li>
      <a href="../articles/HomeworkThree.html">Homework Three</a>
    </li>
    <li>
      <a href="../articles/HomeworkTwo.html">Homework Two</a>
    </li>
    <li>
      <a href="../articles/ProjectOne.html">ProjectOne</a>
    </li>
    <li>
      <a href="../articles/ProjectTwo.html">Project Two</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="09_Tree_of_life_files/header-attrs-2.11/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>GeneticData</h1>
            
      
      
      <div class="hidden name"><code>09_Tree_of_life.Rmd</code></div>

    </div>

    
    
<div id="a-quick-wrap-up-of-the-git-stuff-from-last-time" class="section level2">
<h2 class="hasAnchor">
<a href="#a-quick-wrap-up-of-the-git-stuff-from-last-time" class="anchor"></a>A quick wrap-up of the git stuff from last time</h2>
<p>We’ll install a package called <code>usethis</code>, which contains some helpful programming tools for the weeks ahead:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"usethis"</span><span class="op">)</span></code></pre></div>
<p>We’ll use this to create a personal access token for github. This ensures we don’t have to retype our password every time we need to access GitHub.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/github-token.html">create_github_token</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<p>And we will use this token to access our account via RStudio.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">credentials</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/credentials/reference/set_github_pat.html">set_github_pat</a></span><span class="op">(</span><span class="st">"PAT here"</span><span class="op">)</span></code></pre></div>
<p>Now, refresh your R session.</p>
<p>Once this is done, we can check that this worked like so:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">usethis</span><span class="fu">::</span><span class="fu"><a href="https://usethis.r-lib.org/reference/git_sitrep.html">git_sitrep</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div id="the-tree-of-life" class="section level2">
<h2 class="hasAnchor">
<a href="#the-tree-of-life" class="anchor"></a>The Tree of Life</h2>
<p>First, we need to do some installs.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="kw">install.packages</span>(<span class="st">"devtools"</span>)<span class="dv">1</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">"ropensci/rotl"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">"ropensci/traits"</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="dt">repo=</span><span class="st">'ropensci/phylotaR'</span>, <span class="dt">build_vignettes=</span><span class="ot">TRUE</span>)</span></code></pre></div>
<p>Now, load your packages:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rotl/">rotl</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/phylotaR">phylotaR</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="open-tree" class="section level2">
<h2 class="hasAnchor">
<a href="#open-tree" class="anchor"></a>Open Tree</h2>
<p>First, we’re going to look for some data. Discuss with your neighbor a clade you like and want to build a tree of. For example, perhaps I really like hominids.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">apes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Pongo"</span>, <span class="st">"Pan"</span>, <span class="st">"Gorilla"</span>, <span class="st">"Hoolock"</span>, <span class="st">"Homo"</span><span class="op">)</span>
<span class="op">(</span><span class="va">resolved_names</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tnrs_match_names.html">tnrs_match_names</a></span><span class="op">(</span><span class="va">apes</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   search_string unique_name approximate_match ott_id is_synonym          flags
## 1         pongo       Pongo             FALSE 417949      FALSE               
## 2           pan         Pan             FALSE 417957      FALSE sibling_higher
## 3       gorilla     Gorilla             FALSE 417969      FALSE sibling_higher
## 4       hoolock     Hoolock             FALSE 712902      FALSE               
## 5          homo        Homo             FALSE 770309      FALSE sibling_higher
##   number_matches
## 1              2
## 2              2
## 3              1
## 4              1
## 5              1</code></pre>
<p>Put the names of five genera into the function. If you want, you can use mine below, or pick new ones you prefer.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Martialis"</span>, <span class="st">"Atta"</span>, <span class="st">"Ectatomma"</span>, <span class="st">"Tatuidris"</span>, <span class="st">"Aneuretus"</span>, <span class="st">"Xymmer"</span>, <span class="st">"Aenictus"</span><span class="op">)</span>
<span class="op">(</span><span class="va">resolved_names</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tnrs_match_names.html">tnrs_match_names</a></span><span class="op">(</span><span class="va">ants</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##   search_string unique_name approximate_match  ott_id is_synonym flags
## 1     martialis   Martialis             FALSE  334634      FALSE      
## 2          atta        Atta             FALSE  709964      FALSE      
## 3     ectatomma   Ectatomma             FALSE  162845      FALSE      
## 4     tatuidris   Tatuidris             FALSE  150177      FALSE      
## 5     aneuretus   Aneuretus             FALSE  425343      FALSE      
## 6        xymmer      Xymmer             FALSE 4583488      FALSE      
## 7      aenictus    Aenictus             FALSE  169919      FALSE      
##   number_matches
## 1              1
## 2              1
## 3              1
## 4              1
## 5              1
## 6              1
## 7              1</code></pre>
<p>Try to view them on a tree:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tol_induced_subtree.html">tol_induced_subtree</a></span><span class="op">(</span>ott_ids <span class="op">=</span> <span class="fu"><a href="https://docs.ropensci.org/rotl/reference/taxonomy-methods.html">ott_id</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
Progress [----------------------------------] 0/43 (  0) ?s
Progress [================================] 43/43 (100)  0s
                                                            </code></pre>
<pre><code>## Warning in collapse_singles(tr, show_progress): Dropping singleton nodes
## with labels: mrcaott1682ott35311, mrcaott1682ott34821, mrcaott34821ott950983,
## Aneuretinae ott393261, mrcaott4706ott5525, mrcaott5525ott8038,
## mrcaott8038ott8685, mrcaott8685ott36581, mrcaott8685ott56535,
## mrcaott8685ott72729, mrcaott8685ott86888, mrcaott86888ott129302,
## Attini ott344333, mrcaott114767ott114779, mrcaott114779ott339738,
## mrcaott114779ott116619, mrcaott116619ott148996, mrcaott148996ott1071522,
## mrcaott148996ott150184, mrcaott148996ott339763, mrcaott148996ott409299,
## mrcaott48019ott688185, Ectatomminae ott327970, mrcaott48019ott162057,
## mrcaott7438ott5553505, mrcaott7438ott7439, mrcaott7439ott165217,
## mrcaott7439ott15430, mrcaott15430ott120483, mrcaott15430ott150187,
## mrcaott15430ott169913, mrcaott17752ott98880, mrcaott17752ott150176,
## mrcaott150176ott1037289, mrcaott150176ott4539262, Agroecomyrmecinae ott504610,
## mrcaott98886ott152924, mrcaott152924ott452719, mrcaott152924ott739132,
## mrcaott152924ott503313, mrcaott503313ott928530, mrcaott928530ott6280946,
## Martialinae ott334635</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">tr</span><span class="op">)</span></code></pre></div>
<p><img src="09_Tree_of_life_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<p>I’m a little surprised by this tree.</p>
</div>
<div id="looking-for-data" class="section level2">
<h2 class="hasAnchor">
<a href="#looking-for-data" class="anchor"></a>Looking for data</h2>
<p>We’re going to try to build a tree of these taxa from GenBank Data. Much like accessing NOAA data, we will need to make an API Key. Go <a href="https://www.ncbi.nlm.nih.gov/account/">here</a> and register.</p>
<p>Now that we have that key, we can access NCBI:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/">rentrez</a></span><span class="op">)</span>
<span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="st">"Ectatomma"</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></code></pre></div>
<pre><code>## Entrez search result with 679 hits (object contains 20 IDs and no web_history object)
##  Search term (as translated):  "Ectatomma"[Organism] OR Ectatomma[All Fields]</code></pre>
<p>Holy moley that’s a lot of data. If we do that for every taxon, we’ll end up with a mess. Why don’t we try to filter it a bit.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">Ectatomma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="st">"Ectatomma AND 2018:2019[PDAT]"</span>, retmax <span class="op">=</span> <span class="fl">100</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></code></pre></div>
<p>So there is a good amount of data for our Ectatomma ants. We’re going to download some. And this gets a little tricky. First, I want you to run each line in the loop and get a feel for what each does. What does <code>entrez_fetch</code> do? Waht does <code>cat</code> do? Feel free to look at documentation.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span><span class="op">(</span> <span class="va">seq_start</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">50</span>,<span class="fl">100</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
     <span class="va">recs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"nuccore"</span>, id <span class="op">=</span> <span class="va">Ectatomma</span><span class="op">$</span><span class="va">ids</span>,
                          rettype<span class="op">=</span><span class="st">"fasta"</span>, retmax<span class="op">=</span><span class="fl">50</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">recs</span>, file<span class="op">=</span><span class="st">"Ect.seqs.fasta"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">seq_start</span><span class="op">+</span><span class="fl">49</span>, <span class="st">"sequences downloaded\r"</span><span class="op">)</span>
 <span class="op">}</span></code></pre></div>
<pre><code>## 50 sequences downloaded</code></pre>
<p>Now, let’s look at this file. What do we notice about it? What information is in this file?</p>
<p>Before we move on, we’re going to try one other thing. Try modifying one of the two chunks above (HINT: you will have to rerun both) in order to only get data from the mitochondrial COI gene.</p>
<p>All right, let’s move on to trying to write a function that will return all IDs for all of the ants.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span>
    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ant</span>, <span class="st">" AND "</span>, <span class="st">"2016:2019[PDAT]"</span><span class="op">)</span>
    <span class="va">search_returns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">20</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span>

<span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>Have a look at ?sapply. Talk to a partner and see if you can figure out what sapply does.</p>
<p>So what is the problem with this output? It’s hard to parse - let’s make it easier to pull data for one taxon at a time.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span>
    <span class="va">search_returns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ant</span>, <span class="st">" AND "</span>, <span class="st">"2016:2019[PDAT]"</span><span class="op">)</span>
    <span class="va">search_returns</span><span class="op">[</span><span class="va">ant</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">20</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span>

<span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>There’s one remaining problem, though. Have a look at your outputs. How do you think we can fix this one? Let’s have a look at when papers related to Xymmer were published:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xymmer_papers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pmc"</span>, term<span class="op">=</span><span class="st">"Xymmer"</span>, id <span class="op">=</span> <span class="va">xymmer_papers</span><span class="op">$</span><span class="va">ids</span>, retmax<span class="op">=</span><span class="fl">20</span>, rettype<span class="op">=</span> <span class="st">"text"</span>, api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span></code></pre></div>
<p>Does this give you a hint about something that needs to change to get more data? Make the change in the below code:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">search_year</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ant</span><span class="op">)</span><span class="op">{</span>
    <span class="va">search_returns</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">ant</span>, <span class="st">" AND "</span>, <span class="st">"HINT HINT HINT:2019[PDAT]"</span><span class="op">)</span>
    <span class="va">search_returns</span><span class="op">[</span><span class="va">ant</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"Nucleotide"</span>, term<span class="op">=</span><span class="va">query</span>, retmax<span class="op">=</span><span class="fl">1</span>,api_key <span class="op">=</span><span class="st">"96b569c049fe3d3055c5b747112dfec84308"</span><span class="op">)</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">search_returns</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">ants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html">as.vector</a></span><span class="op">(</span><span class="va">resolved_names</span><span class="op">$</span><span class="va">unique_name</span><span class="op">)</span>

<span class="va">labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">ants</span>,<span class="va">search_year</span>,  USE.NAMES <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<p>Now, we’re going to get the genetic data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">search_seq</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">labels</span><span class="op">)</span><span class="op">{</span>
     <span class="va">recs</span><span class="op">[</span><span class="va">ant</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"nuccore"</span>, id <span class="op">=</span> <span class="va">labels</span>,
                          rettype<span class="op">=</span><span class="st">"fasta"</span>, retmax<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
     <span class="fu"><a href="https://rdrr.io/r/base/cat.html">cat</a></span><span class="op">(</span><span class="va">recs</span>, file<span class="op">=</span><span class="st">"Ants.seqs.fasta"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">recs</span><span class="op">)</span>
<span class="op">}</span>


<span class="va">seq_returns</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">labels</span>, <span class="va">search_seq</span>,  USE.NAMES <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p>What might be a problem with these data? Look carefully at the headers. Now, decide how to fix this. I suggest thinking about a common gene for phylogenetics, COI.</p>
</div>
<div id="just-for-fun" class="section level2">
<h2 class="hasAnchor">
<a href="#just-for-fun" class="anchor"></a>Just for fun</h2>
<p>Take a quick look at this output - it’s a little neat. We’re not going to go into building phylogenetic trees - that’s a topic for Systematics. Here’s a quick look at these data.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"ape"</span><span class="op">)</span></code></pre></div>
<pre><code>## Installing package into '/private/var/folders/54/9kd8nf1x4fnft0ymvb11qmc80000gn/T/RtmpguM06f/temp_libpath356b42006f2'
## (as 'lib' is unspecified)</code></pre>
<pre><code>## 
## The downloaded binary packages are in
##  /var/folders/54/9kd8nf1x4fnft0ymvb11qmc80000gn/T//Rtmp4mo0y2/downloaded_packages</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">coi</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.dna.html">read.dna</a></span><span class="op">(</span><span class="st">"Ants.seqs.fasta"</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by April Wright.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
